mirascope/wandb/prompt.py

```diff
@@ -53,10 +49,66 @@ class HiPrompt(WandbPrompt):
         default_factory=lambda: round(datetime.datetime.now().timestamp() * 1000)
     )
 
-    call_params = BaseCallParams(model="gpt-3.5-turbo-0125")
+    call_params = OpenAICallParams(model="gpt-3.5-turbo-0125")
+
+    def create_with_trace(
+        self, parent: Optional[Trace] = None
+    ) -> tuple[Optional[OpenAIChatCompletion], Trace]:
+        """Creates an OpenAI chat completion and logs it via a W&B `Trace`.
+
+        Args:
+            parent: The parent trace to connect to.
+
+        Returns:
+            A tuple containing the completion and its trace (which has been connected
+                to the parent).
+        """
+        try:
+            completion = super().create()
+            return completion, self._trace(completion, parent)
+        except Exception as e:
+            return None, self._trace_error(e, parent)
+
+    @overload
+    def extract_with_trace(
+        self, schema: Type[BaseTypeT], parent: Optional[Trace], retries: int = 0
+    ) -> tuple[BaseTypeT, Trace]:
+        ...  # pragma: no cover
+
+    @overload
+    def extract_with_trace(
+        self, schema: Type[BaseModelT], parent: Optional[Trace], retries: int = 0
+    ) -> tuple[BaseModelT, Trace]:
+        ...  # pragma: no cover
+
+    @overload
+    def extract_with_trace(
+        self, schema: Callable, parent: Optional[Trace], retries: int = 0
```

Comment:
change all of the `parent: Optional[Trace] = None` to have none as written here as the default